server/src/main/java/org/candlepin/resource/CdnResource.java

```diff
@@ -88,36 +96,110 @@ public void delete(@PathParam("label") String label,
+    public CdnDTO create(
+        @ApiParam(name = "cdn", required = true) CdnDTO cdnDTOInput,
         @Context Principal principal) {
-        Cdn existing = curator.lookupByLabel(cdn.getLabel());
+        Cdn existing = curator.lookupByLabel(cdnDTOInput.getLabel());
         if (existing != null) {
-            throw new BadRequestException(i18n.tr("A CDN with the label {0} already exists", cdn.getLabel()));
+            throw new BadRequestException(i18n.tr(
+                "A CDN with the label {0} already exists", cdnDTOInput.getLabel()));
         }
-        return cdnManager.createCdn(cdn);
+
+        Cdn cndToCreate = new Cdn();
+        this.populateEntity(cndToCreate, cdnDTOInput);
+
+        return this.translator.translate(cdnManager.createCdn(cndToCreate), CdnDTO.class);
     }
 
     @ApiOperation(notes = "Updates a CDN @return a Cdn object", value = "update")
     @PUT
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/{label}")
-    public Cdn update(@PathParam("label") String label,
-        @ApiParam(name = "cdn", required = true) Cdn cdn,
+    public CdnDTO update(@PathParam("label") String label,
+        @ApiParam(name = "cdn", required = true) Cdn cdnDTOInput,
         @Context Principal principal) {
         Cdn existing = verifyAndLookupCdn(label);
-        if (!StringUtils.isBlank(cdn.getName())) {
-            existing.setName(cdn.getName());
+        if (!StringUtils.isBlank(cdnDTOInput.getName())) {
+            existing.setName(cdnDTOInput.getName());
         }
-        if (!StringUtils.isBlank(cdn.getUrl())) {
-            existing.setUrl(cdn.getUrl());
+        if (!StringUtils.isBlank(cdnDTOInput.getUrl())) {
+            existing.setUrl(cdnDTOInput.getUrl());
         }
-        if (cdn.getCertificate() != null) {
-            existing.setCertificate(cdn.getCertificate());
+        if (cdnDTOInput.getCertificate() != null) {
+            existing.setCertificate(cdnDTOInput.getCertificate());
         }
         cdnManager.updateCdn(existing);
-        return existing;
+        return this.translator.translate(existing, CdnDTO.class);
+    }
+
+    /**
+     * Populates the specified entity with data from the provided DTO.
+     * This method will not set the ID field.
+     *
+     * @param entity
+     *  The entity instance to populate
+     *
+     * @param dto
+     *  The DTO containing the data with which to populate the entity
+     *
+     * @throws IllegalArgumentException
+     *  if either entity or dto are null
+     */
+    private void populateEntity(Cdn entity, CdnDTO dto) {
+        if (entity == null) {
+            throw new IllegalArgumentException("the Cdn model entity is null");
+        }
+
+        if (dto == null) {
+            throw new IllegalArgumentException("the Cdn dto is null");
+        }
+
+        if (dto.getName() != null) {
+            entity.setName(dto.getName());
+        }
+
+        if (dto.getLabel() != null) {
```

Comment:
We don't allow setting the label. This appears to be our external ID for this entity.

Reply:
Then who creates the label? CdnResource.create() explicitly says that it expects a unique label, otherwise will not create the Cdn:
```
    public CdnDTO create(
        @ApiParam(name = "cdn", required = true) CdnDTO cdnDTOInput,
        @Context Principal principal) {
        Cdn existing = curator.lookupByLabel(cdnDTOInput.getLabel());
        if (existing != null) {
            throw new BadRequestException(i18n.tr(
                "A CDN with the label {0} already exists", cdnDTOInput.getLabel()));
        }
```

Reply:
Ahh, I misread the diff here and had it in my head we were using it in both update and create. Let me address update specifically, then.

Reply:
I'll remove setting the label in populateEntity, so that I can use it on both update and create methods.